pedalboard/process.h

```diff
@@ -102,19 +113,105 @@ process<float>(const py::array_t<float, py::array::c_style> inputArray,
     throw std::runtime_error("More than two channels received!");
   }
 
-  // Cap the buffer size in use to the size of the input data:
-  bufferSize = std::min(bufferSize, numSamples);
+  juce::AudioBuffer<T> ioBuffer(numChannels, numSamples);
+
+  // Depending on the input channel layout, we need to copy data
+  // differently. This loop is duplicated here to move the if statement
+  // outside of the tight loop, as we don't need to re-check that the input
+  // channel is still the same on every iteration of the loop.
+  switch (inputChannelLayout) {
+  case ChannelLayout::Interleaved:
+    for (unsigned int i = 0; i < numChannels; i++) {
+      T *channelBuffer = ioBuffer.getWritePointer(i);
+      // We're de-interleaving the data here, so we can't use copyFrom.
+      for (unsigned int j = 0; j < numSamples; j++) {
+        channelBuffer[j] = static_cast<T *>(inputInfo.ptr)[j * numChannels + i];
+      }
+    }
+    break;
+  case ChannelLayout::NotInterleaved:
+    for (unsigned int i = 0; i < numChannels; i++) {
+      ioBuffer.copyFrom(
+          i, 0, static_cast<T *>(inputInfo.ptr) + (numSamples * i), numSamples);
+    }
+  }
+
+  return ioBuffer;
+}
+
+template <typename T>
+py::array_t<T> copyJuceBufferIntoPyArray(const juce::AudioBuffer<T> juceBuffer,
+                                         ChannelLayout channelLayout,
+                                         int offsetSamples, int ndim = 2) {
+  unsigned int numChannels = juceBuffer.getNumChannels();
+  unsigned int numSamples = juceBuffer.getNumSamples();
+  unsigned int outputSampleCount =
+      std::max((int)numSamples - (int)offsetSamples, 0);
+
+  // TODO: Avoid the need to copy here if offsetSamples is 0!
+  py::array_t<T> outputArray;
+  if (ndim == 2) {
+    switch (channelLayout) {
+    case ChannelLayout::Interleaved:
+      outputArray = py::array_t<T>({outputSampleCount, numChannels});
+      break;
+    case ChannelLayout::NotInterleaved:
+      outputArray = py::array_t<T>({numChannels, outputSampleCount});
+      break;
```

Comment:
Usually it's a good idea to have a default here with an error throw just in case. Shouldnt be possible theoretically, but you never know.